translator/tocwconfig/totomlconfig/toTomlConfig.go

```diff
@@ -11,11 +11,46 @@ import (
 )
 
 func ToTomlConfig(val interface{}) string {
+	// Process value to ensure integers in arrays are preserved
+	processedVal := processValue(val)
+
 	buf := bytes.Buffer{}
 	enc := toml.NewEncoder(&buf)
-	err := enc.Encode(val)
+	err := enc.Encode(processedVal)
 	if err != nil {
 		log.Panicf("Encode to a valid TOML config fails because of %v", err)
 	}
 	return buf.String()
 }
+
+// ensure integers in arrays are preserved
+func processValue(val interface{}) interface{} {
```

Comment:
shouldn't we have some kind of validation and returning error here. With the current implementation it seems like this function could never fail, is this true?